File: backend/src/main/java/com/labelvie/lablecious/backend/models/entity/Plate.java
package com.labelvie.lablecious.backend.models.entity;


import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Set;

@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "plates")
public class Plate{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(nullable = false)
    private String name;

    @Column
    private String  description;
    @Column
    private String image;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false, referencedColumnName = "id")
    private Category category;

    @OneToMany(mappedBy = "plate", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<Feedback> feedbacks;

    @OneToMany(mappedBy = "plate", cascade = CascadeType.ALL)
    private Set<MenuPlate> menuPlates;

}

File: labelcious_mobile/lib/models/Plate.dart
import 'dart:convert';
import 'package:mobilelabelcious/models/Category.dart';
import 'package:mobilelabelcious/models/Feedback.dart';

List<Plate> plateFromJson(String str) =>
    List<Plate>.from(json.decode(str).map((x) => Plate.fromJson(x)));

String plateToJson(List<Plate> data) =>
    json.encode(List<dynamic>.from(data.map((x) => x.toJson())));

class Plate {
  int? id;
  String? name;
  String? description;
  String? image;
  int? category_id;
  Category? category;
  List<Feedback>? feedbacks;
  int? quantity;

  Plate({
    this.id,
    this.name,
    this.description,
    this.image,
    this.category_id,
    this.category,
    this.feedbacks,
    this.quantity,
  });

  factory Plate.fromJson(Map<String, dynamic> json) => Plate(
        id: json["id"],
        name: json["name"],
        description: json["description"],
        category_id: json["category_id"],
        category: Category.fromJson(json["category"]),
        feedbacks: List<Feedback>.from(
            json["feedbacks"].map((x) => Feedback.fromJson(x))),
        quantity: json["quantity"],
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "name": name,
        "description": description,
        "category_id": category_id,
        "category": category?.toJson(),
        "feedbacks":
            List<dynamic>.from(feedbacks?.map((x) => x.toJson()) ?? []),
        "quantity": quantity,
      };
}
